components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultOIDCClaimsCallbackHandler.java

```diff
@@ -541,15 +544,26 @@ private Map<String, Object> getUserClaimsInOIDCDialect(String spTenantDomain,
         /* For B2B users, the resident organization is available to find the tenant where the user's identity is
         managed. Hence, the correct tenant domain should be used to fetch user claims. */
         if (!StringUtils.equals(userTenantDomain, userResidentTenantDomain)) {
```

Comment:
The code changes we have don could have been minimized if we had organized in this manner.
1 - String userId = authenticatedUser.getUserId();
2 - if (!StringUtils.equals(userTenantDomain, userResidentTenantDomain)) { 
          if (authenticatedUser.isFederatedUser()) {
             userId = resolveUserIdForOrganizationSsoUser(authenticatedUser) 
         }
3 - The rest of the code.

Reply:
Addressed with c202638bd7d45d70c3c4030d1f697325e8fdbc4f